connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -98,6 +98,44 @@ abstract class AvroSuite
     }, new GenericDatumReader[Any]()).getSchema.toString(false)
   }
 
+  def checkUnionStableId(
+    types: List[Schema],
+    expectedSchema: String,
+    fieldsAndRow : Seq[(Any, Row)]): Unit = {
```

Comment:
nit: no space before `:`.

Also, could you explain the type? I think you can just pass the expected DataFrame that would contain the schema and the data.

Reply:
I copied the way SQL schema is generated from test "Complex Union Type". I feel it is easier to write unit test this way and if possible I will maintain it. I will add comments to explain the parameters.